### PR TITLE
[GLUTEN-7581][CORE] Code cleanup for GlutenColumnarRule

### PR DESCRIPTION
**Pull request by beliefer**
## What changes were proposed in this pull request?

This PR proposes to make code cleanup for `GlutenColumnarRule`.

## How was this patch tested?

integration tests

### CODE REVIEW

File: gluten-core/src/main/scala/org/apache/gluten/extension/GlutenColumnarRule.scala [hunk 1]
```diff
@@ -109,16 +77,31 @@ case class GlutenColumnarRule(
...
+        (child, false)
+      case RowToColumnarExec(DummyRowOutputExec(child)) =>
+        (child, true)
+      case DummyColumnarOutputExec(child) =>
+        (child, true)
+      case ColumnarToRowExec(DummyColumnarOutputExec(child)) =>
+        (child, false)
+      case _ =>
+        throw new IllegalStateException(
+          "This should not happen. Please leave a issue at" +
```
Comment (zhztheplayer) [inline gluten-core/src/main/scala/org/apache/gluten/extension/GlutenColumnarRule.scala:99]: Can you help change `a issue` to `an issue` in passing? Thanks.

File: gluten-core/src/main/scala/org/apache/gluten/extension/GlutenColumnarRule.scala [hunk 2]
```diff
@@ -109,16 +77,31 @@ case class GlutenColumnarRule(
...
+    } else {
+      DummyRowOutputExec(plan)
+    }
   }
 
   override def postColumnarTransitions: Rule[SparkPlan] = plan => {
-    val outputsColumnar = OutputsColumnarTester.inferOutputsColumnar(plan)
-    val unwrapped = OutputsColumnarTester.unwrap(plan)
-    val vanillaPlan = Transitions.insertTransitions(unwrapped, outputsColumnar)
+    val (childPlan, outputsColumnar) = plan match {
```
Comment (zhztheplayer) [inline gluten-core/src/main/scala/org/apache/gluten/extension/GlutenColumnarRule.scala:88]: Will renaming `childPlan` to `originalPlan` or similar help user to understand the code?

File: gluten-core/src/main/scala/org/apache/gluten/extension/GlutenColumnarRule.scala [hunk 3]
```diff
@@ -109,16 +77,31 @@ case class GlutenColumnarRule(
...
+      case _ =>
+        throw new IllegalStateException(
+          "This should not happen. Please leave a issue at" +
+            " https://github.com/apache/incubator-gluten.")
+    }
+    val vanillaPlan = Transitions.insertTransitions(childPlan, outputsColumnar)
     val applier = applierBuilder.apply(session)
-    val out = applier.apply(vanillaPlan, outputsColumnar)
-    out
+    applier.apply(vanillaPlan, outputsColumnar)
```
[the remark this answers is not in the document]
Comment (beliefer) [reply gluten-core/src/main/scala/org/apache/gluten/extension/GlutenColumnarRule.scala:104]: Yes. I understood it. We usually avoid this code pattern in Spark. Developers could evaluate the expression (e.g. out) if who want know the value.
Of course, I will respect the code pattern if Gluten has the preference.

[the remark this answers is not in the document]
Comment (zhztheplayer) [reply gluten-core/src/main/scala/org/apache/gluten/extension/GlutenColumnarRule.scala:104]: Let's do the simplification. Just found there is an [alternative way](https://stackoverflow.com/questions/5010362/can-i-find-out-the-return-value-before-returning-while-debugging-in-intellij) while using Intellij IDE. Thanks.

**Review by zhztheplayer — APPROVED** [review]

**Comment by beliefer** [comment]
@zhztheplayer Thank you!

**Pull request merged**